LibreNMS/OS/Cnpilote.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Pmp.php
+ *
+ * Cambium
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2017 Paul Heinrichs
+ * @author     Paul Heinrichs<pdheinrichs@gmail.com>
+ */
+
+namespace LibreNMS\OS;
+
+use LibreNMS\Device\WirelessSensor;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessClientsDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessSnrDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessPowerDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessNoiseFloorDiscovery;
+use LibreNMS\OS;
+
+class Pmp extends OS implements
+    WirelessClientsDiscovery,
+    WirelessSnrDiscovery,
+    WirelessPowerDiscovery,
+    WirelessNoiseFloorDiscovery
+{
+
+    /**
+     * Discover wireless client counts. Type is clients.
+     * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
+     *
+     * @return array Sensors
+     */
+    public function discoverWirelessClients()
+    {
+        $oid = '.1.3.6.1.4.1.17713.22.1.1.1.14.0'; //CAMBIUM-MIB::cambiumAPTotalClients.0
+        return array(
+            new WirelessSensor('clients', $this->getDeviceId(), $oid, 'cnpilot', 1, 'Clients')
+        );
+    }
+
+    /**
+     * Discover wireless SNR.  This is in dB. Type is snr.
+     * Formula: SNR = Signal or Rx Power - Noise Floor
+     * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
+     *
+     * @return array Sensors
+     */
+    public function discoverWirelessSnr()
+    {
+        $oid = '.1.3.6.1.4.1.17713.22.1.3.1.11.0'; //CAMBIUM-MIB::cambiumClientSNR.0
+        return array(
+            new WirelessSensor('snr', $this->getDeviceId(), $oid, 'cnpilot', 1, 'SNR')
+        );
+    }
+
+    /**
+     * Discover wireless tx or rx power. This is in dBm. Type is power.
+     * Returns an array of LibreNMS\Device\Sensor objects that have been discovered
+     *
+     * @return array
+     */
+    public function discoverWirelessPower()
+    {
+        $oid = '.1.3.6.1.4.1.17713.22.1.2.1.8.0'; //CAMBIUM-MIB::cambiumRadioTransmitPower.0
```

Comment:
When I looked a the MIB a bit earlier, I noticed these are table entries.  Could there be more than one point of data returned (multiple radios)?

Do you have a device to test or are you just adding based on the MIB @pheinrichs ?

Reply:
I don't have any devices to test on the cnpilot-e devices. This is purely on the mib